api/csiaddons/v1alpha1/encryptionkeyrotationcronjob_types.go

```diff
@@ -0,0 +1,128 @@
+	// +kubebuilder:validation:Required
+	Spec EncryptionKeyRotationJobSpec `json:"spec,omitempty"`
+}
+
+// EncryptionKeyRotationCronJobSpec defines the desired state of EncryptionKeyRotationCronJob
+type EncryptionKeyRotationCronJobSpec struct {
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:=.+
+	Schedule string `json:"schedule"`
+
+	// Optional deadline in seconds for starting the job if it misses scheduled
+	// time for any reason.  Missed jobs executions will be counted as failed ones.
+	// +kubebuilder:validation:Optional
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`
+
+	// Specifies how to treat concurrent executions of a Job.
+	// Valid values are:
+	// - "Forbid" (default): forbids concurrent runs, skipping next run if
+	//   previous run hasn't finished yet;
+	// - "Replace": cancels currently running job and replaces it
+	//   with a new one
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=Forbid;Replace
+	// +kubebuilder:default:=Forbid
+	ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty"`
+
+	// This flag tells the controller to suspend subsequent executions, it does
+	// not apply to already started executions.  Defaults to false.
+	// +kubebuilder:validation:Optional
+	Suspend *bool `json:"suspend,omitempty"`
+
+	// Specifies the job that will be created when executing a CronJob.
+	// +kubebuilder:validation:Required
+	JobSpec EncryptionKeyRotationJobTemplateSpec `json:"jobTemplate"`
+
+	// The number of successful finished jobs to retain. Value must be non-negative integer.
+	// Defaults to 3.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum=60
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=3
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty"`
+
+	// The number of failed finished jobs to retain. Value must be non-negative integer.
+	// Defaults to 1.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum=60
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=1
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit,omitempty"`
+}
+
+// EncryptionKeyRotationCronJobStatus defines the observed state of EncryptionKeyRotationCronJob
+type EncryptionKeyRotationCronJobStatus struct {
+	// A pointer to currently running job.
+	Active *v1.ObjectReference `json:"active,omitempty"`
+
+	// Information when was the last time the job was successfully scheduled.
+	LastScheduleTime *metav1.Time `json:"lastScheduleTime,omitempty"`
+
+	// Information when was the last time the job successfully completed.
+	LastSuccessfulTime *metav1.Time `json:"lastSuccessfulTime,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+//+kubebuilder:printcolumn:JSONPath=".spec.schedule",name=Schedule,type=string
```

Comment:
either add spaces before `+kubebuilder`, or do not. Don't mix them in the same block please.

Reply:
Resolved, would like to add, kubebuilder now adds spaces by default on v4 layout.